16 Lazy Loading/src/index.tsx

```diff
@@ -2,69 +2,41 @@ import * as React from 'react';
 import * as ReactDOM from 'react-dom';
 import { Router, Route, IndexRoute, hashHistory } from 'react-router';
 import App from './components/app.tsx';
+import * as _AboutPage from "./components/about/aboutPage";
+import * as _MemberPage from "./components/member/memberPage";
+import * as _MembersPage from "./components/members/membersPage";
 
-//Loading single component in one chunk
-const lazyLoadAboutComponent = () => {
-  return {
-      getComponent: (location, callback)=> {
-        require.ensure([], require => {
-          callback(null, require('./components/about/aboutPage')["default"]);
-        }, 'AboutPage');
-      }
-    }
-};
-
-//Loading group of components in one chunk
-const lazyLoadMemberComponent = () => {
-  return {
-      getComponent: (location, callback) => {
-        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
-          callback(null, require('./components/member/memberPage')["default"]);
-        }, 'MemberComponents');
-      }
-    }
-};
+type LoadCallback = (error: any, component: React.ComponentClass<any>) => void;
 
-const lazyLoadMembersComponent = () => {
-  return {
-      getComponent: (location, callback) => {
-        require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
-          callback(null, require('./components/members/membersPage')["default"]);
-        }, 'MemberComponents');
-      }
-    }
+function loadAboutPage(location: any, callback: LoadCallback) {
+  require.ensure(
+    [],
+    () => callback(null, (require("./components/about/aboutPage") as typeof _AboutPage).default),
+    "AboutPage");
 };
 
-//Second approach to load group of components in one chunk could be:
-/*
-    const lazyLoadMemberAreaPage = (pageName) => {
-      return {
-          getComponent: (location, callback) => {
-            require.ensure(['./components/member/memberPage', './components/members/membersPage'], require => {
-              switch(pageName) {
-                case 'member':
-                  callback(null, require('./components/member/memberPage')["default"]);
-                  break;
+function loadMemberPage(location: any, callback: LoadCallback) {
+  require.ensure(
+    [],
+    () => callback(null, (require("./components/member/memberPage") as typeof _MemberPage).default),
+    "MemberPages");
+}
 
-                case 'members':
-                  callback(null, require('./components/members/membersPage')["default"]);
-                  break;
-              }
-            }, 'MemberComponents');
-          }
-        }
-    };
-*/
+function loadMembersPage(location: any, callback: LoadCallback) {
+  require.ensure(
+    [],
+    () => callback(null, (require("./components/members/membersPage") as typeof _MembersPage).default),
```

Comment:
From my PR description: *not because of the underscore, that's just a notation, but because they are only used as types*. I use the underscore just for readability.
See the [handbook](http://www.typescriptlang.org/docs/handbook/modules.html#optional-module-loading-and-other-advanced-loading-scenarios).